pytorch_lightning/strategies/ddp.py

```diff
@@ -144,6 +151,7 @@ def _is_single_process_single_device(self) -> bool:
 
     def setup_environment(self) -> None:
         # start the other scripts
+        assert self.cluster_environment is not None
```

Comment:
style-wise, `assert` is used for tests. since asserts can be disabled, explicitly raise an error like this:
```suggestion
        if self.cluster_environment is None:
            raise AssertionError("...")
```

Reply:
@ananthsub we prefer plain `assert`s for nullity checks that we know will not fail.

Asserts being disabled would be good in that case since we don't want them to run, they are just for there for `mypy`.

Since we rely a lot on hooks, any variable that's optional will need to be asserted in each hook even if we know it was created in an initial hook (`setup` for example)